assert/assert_test.go

```diff
@@ -0,0 +1,221 @@
+package assert_test
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"github.com/madlambda/spells/assert"
+)
+
+func TestPartial(t *testing.T) {
+	type testcase struct {
+		name   string
+		obj    interface{}
+		target interface{}
+		fail   bool
+	}
+
+	for _, tc := range []testcase{
+		{
+			name:   "same numbers",
+			obj:    1,
+			target: 1,
+		},
+		{
+			name:   "different int64 numbers - bigger than int32",
+			obj:    int64(math.MaxInt32 + 1),
+			target: int64(math.MaxInt32 + 2),
+			fail:   true,
+		},
+		{
+			name:   "same uint64 numbers - bigger than int64",
+			obj:    uint64(9223372036854775807),
+			target: uint64(9223372036854775807 + 1),
+			fail:   true,
+		},
+		{
+			name:   "numbers mismatch",
+			obj:    1,
+			target: 0,
+			fail:   true,
+		},
+		{
+			name:   "same bool",
+			obj:    true,
+			target: true,
+		},
+		{
+			name:   "bool mismatch",
+			obj:    true,
+			target: false,
+			fail:   true,
+		},
+		{
+			name:   "same floats",
+			obj:    1.2,
+			target: 1.2,
+		},
+		{
+			name:   "different floats",
+			obj:    1.3,
+			target: 1.2,
+			fail:   true,
+		},
+		{
+			name:   "same empty struct",
+			obj:    struct{}{},
+			target: struct{}{},
+		},
+		{
+			name: "different number of fields",
+			obj: struct {
+				A int
+			}{},
+			target: struct{}{},
+			fail:   true,
+		},
+		{
+			name: "different field types",
+			obj: struct {
+				A int
+			}{},
+			target: struct {
+				B string
+			}{},
+			fail: true,
+		},
+		{
+			name: "same struct types different field names",
+			obj: struct {
+				A int
+			}{},
+			target: struct {
+				B int
+			}{},
+			fail: true,
+		},
+		{
+			name: "same struct value",
+			obj: struct {
+				A int
+			}{1},
+			target: struct {
+				A int
+			}{1},
+		},
+		{
+			name: "different struct field value",
+			obj: struct {
+				A string
+			}{"test"},
+			target: struct {
+				A string
+			}{"test2"},
+			fail: true,
+		},
+		{
+			obj: struct {
+				A string
+			}{"test"},
+			target: struct {
+				A string
+			}{"test"},
+			name: "same struct field value",
+		},
```

Comment:
```suggestion
		{
			name: "same struct string value",
			obj: struct {
				A string
			}{"test"},
			target: struct {
				A string
			}{"test"},
		},
```

This seems to be similar to the int test done before, so I imagine we are covering different types.